patch/1.21.4/ngx_lua-ssl_session_hostname.patch

```diff
@@ -0,0 +1,51 @@
+diff --git src/ngx_http_lua_ssl_certby.c src/ngx_http_lua_ssl_certby.c
+index b8e70dde..c3bfc790 100644
+--- src/ngx_http_lua_ssl_certby.c
++++ src/ngx_http_lua_ssl_certby.c
+@@ -870,6 +870,46 @@ ngx_http_lua_ffi_ssl_server_name(ngx_http_request_t *r, char **name,
+ }
+ 
+ 
++int
++ngx_http_lua_ffi_ssl_session_hostname(ngx_http_request_t *r, char **name,
++    size_t *namelen, char **err)
++{
++    ngx_ssl_conn_t          *ssl_conn;
++
++    if (r->connection == NULL || r->connection->ssl == NULL) {
++        *err = "bad request";
++        return NGX_ERROR;
++    }
++
++    ssl_conn = r->connection->ssl->connection;
++    if (ssl_conn == NULL) {
++        *err = "bad ssl conn";
++        return NGX_ERROR;
++    }
++
++#if (defined(TLS1_3_VERSION)                                                   \
```

Comment:
Will this function be limited to TLSv1.3? That doesn't seem to be evident from its name and description.

Where did you find the following explanation, the OpenSSL documentation does not mention this information.

Reply:
also come from nginx's fix: https://github.com/nginx/nginx/commit/46b9f5d389447b3b822ea71f5ac86ebc316c2975